include/dr/shp/copy.hpp

```diff
@@ -75,19 +75,24 @@ template <std::forward_iterator InputIt, lib::distributed_iterator OutputIt>
   requires __detail::is_syclmemcopyable<std::iter_value_t<InputIt>,
                                         std::iter_value_t<OutputIt>>
 cl::sycl::event copy_async(InputIt first, InputIt last, OutputIt d_first) {
-  auto segments = lib::ranges::segments(d_first);
-  auto segment = std::begin(segments);
+  auto &&segments = lib::ranges::segments(d_first);
+  auto segment_iter = rng::begin(segments);
```

Comment:
Could you please explain the difference and why to use one over the other?

Reply:
I see an [article](https://stackoverflow.com/questions/62183286/what-is-the-difference-between-stdrangesbegin-and-stdbegin) about it. I'll read.